averbis/core/_rest_client.py

```diff
@@ -1891,11 +1893,24 @@ def _reprocess(self, project_name: str, process_name: str, document_source_name:
         return None
 
     @staticmethod
-    def __handle_error(response):
-        if response["errorMessages"] is None or len(response["errorMessages"]) == 0:
-            return
-
-        raise Exception("Unable to perform request: " + ", ".join(response["errorMessages"]))
+    def __handle_error(raw_response):
+        error_msg = f"Client request failed with status code {raw_response.status_code}."
+        try:
+            response = raw_response.json()
+
+            # Accessing an endpoint that is in the subdomain of the platform, but which does not exist,
+            # returns a general servlet with a field "message"
+            if "message" in response and response["message"] is not None:
+                error_msg += f"\nError message is: {response['message']}"
+
+            # Accessing an existing endpoint that has an error, returns its error in "errorMessages"
+            if "errorMessages" in response and response["errorMessages"] is not None:
+                error_msg += f'\nError message is: {", ".join(response["errorMessages"])}'
+        except JSONDecodeError:
+            # This is the case where we received a response (with a status_code), but no valid json.
```

Comment:
In this case we should probably still include the server response in the error message because it may include an additional error string.

Reply:
Thanks. Dealt with it in the latest commit below.